com.unity.render-pipelines.high-definition/Runtime/Debug/InstancedProbeShader.shader

```diff
@@ -0,0 +1,97 @@
+Shader "Hidden/HDRP/InstancedProbeShader"
+{
+    Properties
+    {
+    }
+
+    SubShader
+    {
+        Tags { "RenderType" = "Opaque" }
+        LOD 100
+        Pass
+        {
+            CGPROGRAM
+            #pragma vertex vert
+            #pragma fragment frag
+            #pragma multi_compile_instancing
+            #include "UnityCG.cginc"
+
+            uniform int _ShadingMode;
+            uniform float _Exposure;
+            uniform float _ProbeSize;
+            uniform float4 _Color;
+
+            struct appdata
+            {
+                float4 vertex : POSITION;
+                float3 normal : NORMAL;
+                UNITY_VERTEX_INPUT_INSTANCE_ID
+            };
+
+            struct v2f
+            {
+                float4 vertex : SV_POSITION;
+                float3 normal : TEXCOORD1;
+                UNITY_VERTEX_INPUT_INSTANCE_ID
+            };
+
+            UNITY_INSTANCING_BUFFER_START(Props)
+                UNITY_DEFINE_INSTANCED_PROP(float4, _R)
+                UNITY_DEFINE_INSTANCED_PROP(float4, _G)
+                UNITY_DEFINE_INSTANCED_PROP(float4, _B)
+                UNITY_DEFINE_INSTANCED_PROP(float4, _Validity)
+            UNITY_INSTANCING_BUFFER_END(Props)
+
+            v2f vert(appdata v)
+            {
+                v2f o;
+
+                UNITY_SETUP_INSTANCE_ID(v);
+                UNITY_TRANSFER_INSTANCE_ID(v, o);
+
+                o.vertex = UnityObjectToClipPos(v.vertex * _ProbeSize);
+                o.normal = UnityObjectToWorldNormal(v.normal);
+
+                return o;
+            }
+
+            float3 evalSH(float3 normal, float4 SHAr, float4 SHAg, float4 SHAb)
+            {
+                float4 normalPadded = float4(normal, 1);
```

Comment:
not sure why you don't reuse the functoin provided in EntityLighting.hlsl?

Reply:
To answer this and the before, @thefranke  lifted this code from another project altogether. We indeed need to rewrite it anyway to be more SRP complaint.

Reply:
This wasn't considered a blocker for opening the PR though.